✨ feat: 어드민 공고 목록 조회 및 태그 검색/필터링 적용

Jiwoong0322 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #176 
- 작업 요약: 어드민 공고 목록 조회 기능 개발

## 📄 상세 내용
- [ ] 관리자용 구인 공고 목록 조회 API 구현 GET /api/v1/admin/recruitments)
- [ ] 관리자(IsAdminUser)만 접근 가능하도록 권한 설정
- [ ] 공고명 검색, 상태 및 태그 기반의 필터링 기능 추가 (CustomRecruitmentFilter 구현)
- [ ] 쿼리 파라미터를 이용한 정렬 기능 추가 (최신순, 조회수, 북마크순)
- [ ] Limit-Offset 기반 페이지네이션 적용
- [ ] 관련 기능에 대한 단위 테스트 코드 작성

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Jiwoong0322 committed:
✨ feat: 어드민 공고 목록 조회 및 태그 검색/필터링 적용

관리자용 구인 공고 목록 조회 API 구현 GET /api/v1/admin/recruitments)
관리자(IsAdminUser)만 접근 가능하도록 권한 설정
공고명 검색, 상태 및 태그 기반의 필터링 기능 추가 (CustomRecruitmentFilter 구현)
쿼리 파라미터를 이용한 정렬 기능 추가 (최신순, 조회수, 북마크순)
Limit-Offset 기반 페이지네이션 적용
관련 기능에 대한 단위 테스트 코드 작성

관련 이슈: #176

Meoyoug commented on apps/recruitments/managers/managers_list.py:
```diff
@@ -13,13 +13,13 @@
         return self.filter(is_closed=False)
 
-    def order_last(self) -> RecruitmentListQuerySet:
+    def order_last(self) -> Self:
```
sort_by_latest 처럼 정확한 명명을 사용해주시면 좋을 것 같아요

Meoyoug replied:
sort_by_latest 처럼 정확한 명명을 사용해주시면 좋을 것 같아요

Jiwoong0322 committed:
♻️ refactor: 함수명 변경

 - managers_list.py의 order_last를 sort_by_latest로 변경
 - services_list.py에서 참조하던 order_last를 sort_by_latest로 변경

관련 이슈: #176
